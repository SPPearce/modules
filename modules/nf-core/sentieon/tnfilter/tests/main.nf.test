nextflow_process {

    name "Test Process SENTIEON_TNFILTER"
    config "./nextflow.config"
    script "../main.nf"
    process "SENTIEON_TNFILTER"
   
    tag "modules"
    tag "modules_nfcore"
    tag "sentieon"
    tag "sentieon/tnfilter"
   
    
    
    test("test_sentieon_tnfilter_base") {
        
        when {
            params {
                module_args = {"${meta.id}.filtered"}
            }
            process {
                """
                input[0] = [
                    [id:'test'],// meta map
                    file(params.niche_test['tnfilter']['vcf'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['tbi'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['stats'],checkIfExists:true),
                    [],
                    [],
                    []
                    ]
                input[1] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                    ]
                input[2] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    
    test("test_sentieon_tnfilter_with_files") {
        
        when {
            params {
                module_args = {"${meta.id}.filtered"}
            }
            process {
                """
                input[0] = [
                    [id:'test'],// meta map
                    file(params.niche_test['tnfilter']['vcf'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['tbi'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['stats'],checkIfExists:true),
                    [file(params.niche_test['tnfilter']['contamination_data'],checkIfExists:true)],
                    [file(params.niche_test['tnfilter']['segments'],checkIfExists:true)],
                    [file(params.niche_test['tnfilter']['orientation_data'],checkIfExists:true)]
                    ]
                input[1] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                    ]
                input[2] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    
    test("test_sentieon_tnfilter_base_stubs") {
        
        when {
            params {
                module_args = {"${meta.id}.filtered"}
            }
            process {
                """
                input[0] = [
                    [id:'test'],// meta map
                    file(params.niche_test['tnfilter']['vcf'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['tbi'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['stats'],checkIfExists:true),
                    [],
                    [],
                    []
                    ]
                input[1] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                    ]
                input[2] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    
    test("test_sentieon_tnfilter_base -- stub") {
        options '-stub'
        when {
            params {
                module_args = {"${meta.id}.filtered"}
            }
            process {
                """
                input[0] = [
                    [id:'test'],// meta map
                    file(params.niche_test['tnfilter']['vcf'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['tbi'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['stats'],checkIfExists:true),
                    [],
                    [],
                    []
                    ]
                input[1] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                    ]
                input[2] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    
    test("test_sentieon_tnfilter_with_files -- stub") {
        options '-stub'
        when {
            params {
                module_args = {"${meta.id}.filtered"}
            }
            process {
                """
                input[0] = [
                    [id:'test'],// meta map
                    file(params.niche_test['tnfilter']['vcf'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['tbi'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['stats'],checkIfExists:true),
                    [file(params.niche_test['tnfilter']['contamination_data'],checkIfExists:true)],
                    [file(params.niche_test['tnfilter']['segments'],checkIfExists:true)],
                    [file(params.niche_test['tnfilter']['orientation_data'],checkIfExists:true)]
                    ]
                input[1] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                    ]
                input[2] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    
    test("test_sentieon_tnfilter_base_stubs -- stub") {
        options '-stub'
        when {
            params {
                module_args = {"${meta.id}.filtered"}
            }
            process {
                """
                input[0] = [
                    [id:'test'],// meta map
                    file(params.niche_test['tnfilter']['vcf'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['tbi'],checkIfExists:true),
                    file(params.niche_test['tnfilter']['stats'],checkIfExists:true),
                    [],
                    [],
                    []
                    ]
                input[1] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                    ]
                input[2] = [[id:'genome'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}