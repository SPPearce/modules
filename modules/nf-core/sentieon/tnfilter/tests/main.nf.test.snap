{
    "vcf, index, stats only - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,8348776c1a3431dd72a7b64665893861"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "vcf_tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8348776c1a3431dd72a7b64665893861"
                ]
            },
            {
                "SENTIEON_TNFILTER": {
                    "sentieon": 202308.03
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-21T13:04:40.956375257"
    },
    "vcf, index, stats only": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz:md5,06b7f4351e9742ae445b0e56ff9b5529"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.tbi:md5,1812f6c608099dd7131d08472938cf13"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.stats:md5,0afebca9cbfe6a666ddf277fed2a6a53"
                    ]
                ],
                "3": [
                    "versions.yml:md5,8348776c1a3431dd72a7b64665893861"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.stats:md5,0afebca9cbfe6a666ddf277fed2a6a53"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz:md5,06b7f4351e9742ae445b0e56ff9b5529"
                    ]
                ],
                "vcf_tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.tbi:md5,1812f6c608099dd7131d08472938cf13"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8348776c1a3431dd72a7b64665893861"
                ]
            },
            {
                "SENTIEON_TNFILTER": {
                    "sentieon": 202308.03
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-21T13:03:45.479119177"
    },
    "with contamination/orientation data": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz:md5,ae438563a100c831fc668d34e17cf092"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.tbi:md5,d92ffa6aabc029aa2092b31b29f2b5e9"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.stats:md5,a3874025218d6540b7c9798b4ef64c59"
                    ]
                ],
                "3": [
                    "versions.yml:md5,8348776c1a3431dd72a7b64665893861"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.stats:md5,a3874025218d6540b7c9798b4ef64c59"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz:md5,ae438563a100c831fc668d34e17cf092"
                    ]
                ],
                "vcf_tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.vcf.gz.tbi:md5,d92ffa6aabc029aa2092b31b29f2b5e9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8348776c1a3431dd72a7b64665893861"
                ]
            },
            {
                "SENTIEON_TNFILTER": {
                    "sentieon": 202308.03
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-21T13:04:13.408647825"
    }
}